homeassistant/components/godice/__init__.py

```diff
@@ -0,0 +1,68 @@
+"""The godice integration."""
+
+from __future__ import annotations
+
+import logging
+
+import bleak
+import godice
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_SHELL, DATA_DICE, DATA_DISCONNECTED_BY_REQUEST_FLAG, DOMAIN
+
+PLATFORMS = [Platform.SENSOR]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up godice from a config entry."""
+    _LOGGER.debug("Setup started")
+
+    def on_disconnect_callback(_ble_data):
+        _LOGGER.debug("on_disconnect_callback called")
+        is_disconnected_by_request = hass.data[DOMAIN][entry.entry_id][
+            DATA_DISCONNECTED_BY_REQUEST_FLAG
+        ]
+        if not is_disconnected_by_request:
+            hass.create_task(hass.config_entries.async_reload(entry.entry_id))
```

Comment:
Why do we reload the integration on disconnect, doesn't the library reconnect?

Reply:
initially I used custom reconnect logic for the integration but then found out that Home Assistant has integration [reload-retry incorporated](https://developers.home-assistant.io/docs/integration_setup_failures?_highlight=configentrynotready#integrations-using-async_setup_entry) so decided to use it to keep the integration code focused and small